tests/3d_examples/test_3d_dambreak/src/Dambreak.cpp

```diff
@@ -69,9 +70,11 @@ class WallBoundary : public SolidBody
 		Vecd halfsize_outer(0.5 * DL + BW, 0.5 * DH + BW, 0.5 * DW + BW);
 		Vecd translation_wall(0.5 * DL, 0.5 * DH, 0.5 * DW);
 		Vecd halfsize_inner(0.5 * DL, 0.5 * DH, 0.5 * DW);
-		body_shape_ = new ComplexShape(body_name);
-		body_shape_->addBrick(halfsize_outer, resolution, translation_wall, ShapeBooleanOps::add);
-		body_shape_->addBrick(halfsize_inner, resolution, translation_wall, ShapeBooleanOps::sub);
+		
+		ComplexShapeTriangleMesh *mesh = new ComplexShapeTriangleMesh();
+		mesh->addBrick(halfsize_outer, resolution, translation_wall, ShapeBooleanOps::add);
+		mesh->addBrick(halfsize_inner, resolution, translation_wall, ShapeBooleanOps::sub);
+		body_shape_ = new ComplexShape(mesh);
```

Comment:
any clean up? unique ptr?

Reply:
body_shape_ is a member in SPHBody class.
I think I have to ask some inputs from @Xiangyu-Hu